hack/build-kubemacpool-podman.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+if [ -z "$ARCH" ] || [ -z "$PLATFORMS" ] || [ -z "$KUBEMACPOOL_IMAGE_TAGGED" ] || [ -z "$KUBEMACPOOL_IMAGE_GIT_TAGGED" ]; then
+    echo "Error: ARCH, PLATFORMS, KUBEMACPOOL_IMAGE_TAGGED, and KUBEMACPOOL_IMAGE_GIT_TAGGED must be set."
+    exit 1
+fi
+
+IFS=',' read -r -a PLATFORM_LIST <<< "$PLATFORMS"
+
+# Remove any existing manifest and image
+podman manifest rm "${KUBEMACPOOL_IMAGE_TAGGED}" 2>/dev/null || true
+podman manifest rm "${KUBEMACPOOL_IMAGE_GIT_TAGGED}" 2>/dev/null || true
```

Comment:
Podman manifest creation will fail if it already exists on the local setup, and it might not be available on the first run. However, it will be available on the second run. The reason for this approach is to ensure that any previously tagged images, even if they didn’t have an associated manifest, are cleaned up. While podman manifest rm won't affect anything if the manifest doesn't exist, to be on safer side we are just ensure there are no leftover artifacts from previous builds.

Additionally, a command has been added to remove dangling images with a <none> tag. This is necessary because podman manifest rm does not remove all associated images. I think this happens because the manifest itself is not the owner of the actual image layers. The images remain as separate platform-specific layers that were pulled or built for the manifest, and removing the manifest doesn't automatically clean up those individual images.

Example:
$ podman images
REPOSITORY                                   TAG         IMAGE ID      CREATED             SIZE
`<none>`                                       `<none>`      b0da9c809b75  53 seconds ago      157 MB
`<none>`                                       `<none> `     3b538855820b  About a minute ago  1.44 GB
`<none>`                                       `<none>`      c3c20eb0721f  3 minutes ago       155 MB
`<none>`                                       `<none> `     78ee518a1726  3 minutes ago       1.43 GB
`<none>`                                       `<none>`      60a174d152ab  5 minutes ago       155 MB
`<none>`                                       `<none>`      253d10dc0021  5 minutes ago       1.44 GB
icr.io/kubevirt/kubemacpool                  ap          63bb3b8a57  6 minutes ago       1.53 kB
icr.io/kubevirt/kubemacpool                  latest      63bb3b8a57  6 minutes ago       1.53 kB

Let me know if that helps clarify!

Reply:
it does, thanks!